### ISSUE TITLE
Eliminate support for out-of-container build, test, etc.

### ISSUE THREAD

**Issue by krancour** [kubernetes-incubator/service-catalog#227]
Currently, make targets involving dependency resolution via glide, compiling, testing etc. can optionally be executed in or out-of container. (Set env var `DOCKER=1` to run in-container.)

As discussed in #219, I'd like us to consider eliminating support for out-of-container builds.

Utilizing in-container builds only minimizes the tooling that any contributor must have access to (go, glide, etc. and specific versions thereof) and maximizes the consistency of the build/test environment from one contributor to the next.

The tangible benefit of eliminating support for out-of-container build, test, etc. is reduced complexity for many of our make targets.

cc @duglin 

Assigning this to myself (because I care about it), but let's all feel free to discuss before I take any action.

**Comment by duglin** [comment]
As I've indicated in the past, I'm +1 on this. While I know there is some overhead involved, I've tried to keep the performance impact to our build to a minimum when running in Docker.  To me the idea that someone can take a machine with nothing but Docker+git+make and build our stuff is a huge UX benefit to me.

**Comment by krancour** [comment]
>  To me the idea that someone can take a machine with nothing but Docker+git+make and build our stuff is a huge UX benefit to me.

Couldn't say it any better, @duglin.

At Deis, we debated this for quite some time before we switched to containerized builds. In the end, we concluded that git, make, and docker were all more ubiquitous than go and glide. A build enabled only by tools that a brand new contributor was most likely to already have was viewed as a huge gain in terms of lowering the barriers to contribution.
